dftimewolf/lib/preflights/sanity_checks.py

```diff
@@ -0,0 +1,62 @@
+"""Sanity checks for recipes."""
+from __future__ import print_function
+
+import datetime
+
+from dftimewolf.lib import module
+from dftimewolf.lib.errors import DFTimewolfError
+from dftimewolf.lib.modules import manager as modules_manager
+
+
+class SanityChecks(module.PreflightModule):
+  """Verifies logic of parameters and fails fast on issues."""
+
+  def __init__(self, state, name=None, critical=False):
+    super(SanityChecks, self).__init__(
+        state, name=name, critical=critical)
+    self.startdate = None
+    self.enddate = None
+    self.dateformat = None
+
+
+  def SetUp(self, startdate=None, enddate=None, dateformat=None):  # pylint: disable=arguments-differ
+    """Sets up a SanityChecks instance.
+
+    Args:
+      startdate (string): A start date of a timespan
+      enddate (string): An end date of a timespan
+      dateformat (string): A date format suitable for datetime.strptime()
+    """
+    self.startdate = startdate
+    self.enddate = enddate
+    self.dateformat = dateformat
+
+  def Process(self):
+    """Test whether the values we've received are sane."""
+
+    try:
+      if (self.startdate and self.enddate and self.dateformat):
+        self._AreDatesValid()
+    except DFTimewolfError:  # We don't need the extra stacktrace here
+      return
```

Comment:
If you catch the `DFTimewolfError` here, as far as the scheduler is concerned the module will be marked as "successfully completed". We want the exception to propagate up to the state where it will [be caught](https://github.com/log2timeline/dftimewolf/blob/master/dftimewolf/lib/state.py#L175) and the recipe flow will be interrupted. So no need to catch it here.

Reply:
Yeah, duly noted, and updated. It wold be nice if there was a way to catch the logic error, but not print the stacktrace in cases this module correctly handles.

Reply:
Well you can raise any other error (`RuntimeError` being quite generic but easy enough to add) in your function and catch that one.